File: BitmapEditorP/Bitmap.py
class Bitmap():
    def __init__(self, br, h):
        self.Breedte = br
        self.Hoogte = h
        self.vakjes = [ [0 for y in range(0,h) ] for x in range(0,br) ]                 

    def veranderKleur(self, x, y, b):
        self.vakjes[x][y] = b
    
    def vraagKleur(self, x, y):
        return self.vakjes[x][y]

    def combineer(self, ander, comb):
        for x in range(0,self.Breedte):
            for y in range(0,self.Hoogte):
                self.veranderKleur(x, y, comb(self.vraagKleur(x,y), ander.vraagKleur(x,y)))

    def Kopieer(self, ander):
        self.combineer(ander, lambda a,b: b )

    def Left(self):
        for y in range(0, self.Hoogte):
            for x in range(1,self.Breedte):
                self.veranderKleur(x-1, y, self.vraagKleur(x,y))
            self.veranderKleur(self.Breedte-1, y, False)

    def Right(self):
        for y in range(0, self.Hoogte):
            for x in range(self.Breedte-1, 0, -1):
                self.veranderKleur(x, y, self.vraagKleur(x-1,y))
            self.veranderKleur(0, y, False)

    def Up(self):
        for x in range(0, self.Breedte):
            for y in range(1,self.Hoogte):
                self.veranderKleur(x, y-1, self.vraagKleur(x,y))
            self.veranderKleur(x, self.Hoogte-1, False)

    def Down(self):
        for x in range(0, self.Breedte):
            for y in range(self.Hoogte-1, 0, -1):
                self.veranderKleur(x, y, self.vraagKleur(x,y-1))
            self.veranderKleur(x, 0, False)

    def Clear(self):
        self.combineer(self, lambda a,b: False )

    def Invert(self):
        self.combineer(self, lambda a,b: not a )

    def Bold(self):
        ander = Bitmap(self.Breedte, self.Hoogte)
        ander.Kopieer(self)
        ander.Left()
        self.combineer(ander, lambda a, b: a or b)
        ander.Kopieer(self)
        ander.Down()
        self.combineer(ander, lambda a, b: a or b)

    def Outline(self):
        ander = Bitmap(self.Breedte, self.Hoogte)
        ander.Kopieer(self)
        ander.Left()
        ander.Down()
        self.combineer(ander, lambda a, b: a != b)

    def Life(self):
        old = Bitmap(self.Breedte, self.Hoogte)
        old.Kopieer(self)
        for y in range(0,self.Hoogte):
            for x in range(0,self.Breedte):
                n = old.Buren(x,y)
                self.veranderKleur(x, y, n==3 or (old.vraagKleur(x,y) and n==2))

    def Buren(self, x, y):
        x0 = x-1
        if x0<0: x0 += self.Breedte
        y0 = y-1
        if y0<0: y0 += self.Hoogte
        x1 = x+1
        if x1>=self.Breedte: x1 -= self.Breedte
        y1 = y+1
        if y1>=self.Hoogte: y1 -= self.Hoogte
        n = 0
        if self.vraagKleur(x0,y0): n+=1
        if self.vraagKleur(x ,y0): n+=1
        if self.vraagKleur(x1,y0): n+=1
        if self.vraagKleur(x0,y): n+=1
        if self.vraagKleur(x1,y): n+=1
        if self.vraagKleur(x0,y1): n+=1
        if self.vraagKleur(x ,y1): n+=1
        if self.vraagKleur(x1,y1): n+=1
        return n

File: BitmapEditorC/Bitmap.cs
using System;

delegate bool Functie(bool x, bool y);

public class Bitmap
{
    private bool[,] vakjes;

    public Bitmap(int br, int h)
    {   vakjes = new bool[br, h];
    }
    public Bitmap(Bitmap ander)
    {   vakjes = new bool[ander.Breedte, ander.Hoogte];
        Kopieer(ander);
    }
    public int Breedte
    {   get { return vakjes.GetLength(0); }
    }
    public int Hoogte
    {   get { return vakjes.GetLength(1); }
    }
    public void veranderKleur(int x, int y, bool b)
    {   vakjes[x, y] = b;
    }
    public bool vraagKleur(int x, int y)
    {   return vakjes[x, y];
    }
    private void combineer(Bitmap ander, Functie comb)
    {   for (int x = 0; x < Breedte; x++)
            for (int y = 0; y < Hoogte; y++)
                veranderKleur(x, y, comb(vraagKleur(x, y), ander.vraagKleur(x, y)));
    }
    private void Kopieer(Bitmap ander)
    {   combineer(ander, (a, b) => b);
    }
    public void Clear()
    {   combineer(this, (a, b) => false);
    }
    public void Invert()
    {   combineer(this, (a, b) => !a);
    }
    public void Bold()
    {   Bitmap ander = new Bitmap(this);
        ander.Left();
        combineer(ander, (a, b) => a || b);
        ander.Kopieer(this);
        ander.Down();
        combineer(ander, (a, b) => a || b);
    }
    public void Outline()
    {   Bitmap ander = new Bitmap(this);
        ander.Left();
        ander.Down();
        combineer(ander, (a, b) => a != b);
    }
    public void Left()
    {   for (int y = 0; y < Hoogte; y++)
        {
            for (int x = 1; x < Breedte; x++)
                veranderKleur(x - 1, y, vraagKleur(x, y));
            veranderKleur(Breedte - 1, y, false);
        }
    }
    public void Right()
    {   for (int y = 0; y < Hoogte; y++)
        {
            for (int x = Breedte - 1; x > 0; x--)
                veranderKleur(x, y, vraagKleur(x - 1, y));
            veranderKleur(0, y, false);
        }
    }
    public void Up()
    {   for (int x = 0; x < Breedte; x++)
        {
            for (int y = 1; y < Hoogte; y++)
                veranderKleur(x, y - 1, vraagKleur(x, y));
            veranderKleur(x, Hoogte - 1, false);
        }
    }
    public void Down()
    {   for (int x = 0; x < Breedte; x++)
        {
            for (int y = Hoogte - 1; y > 0; y--)
                veranderKleur(x, y, vraagKleur(x, y - 1));
            veranderKleur(x, 0, false);
        }
    }
    public void Life()
    {   Bitmap old = new Bitmap(this);
        for (int y = 0; y < Hoogte; y++)
            for (int x = 0; x < Breedte; x++)
            {
                int n = old.buren(x, y);
                veranderKleur(x, y, n == 3 || (old.vraagKleur(x, y) && n == 2));
            }
    }
    private int buren(int x, int y)
    {   int x0 = x - 1; if (x0 < 0) x0 += Breedte;
        int y0 = y - 1; if (y0 < 0) y0 += Hoogte;
        int x1 = x + 1; if (x1 >= Breedte) x1 -= Breedte;
        int y1 = y + 1; if (y1 >= Hoogte) y1 -= Hoogte;
        int n = 0;
        if (vraagKleur(x0, y0)) n++;
        if (vraagKleur(x, y0)) n++;
        if (vraagKleur(x1, y0)) n++;
        if (vraagKleur(x0, y)) n++;
        if (vraagKleur(x1, y)) n++;
        if (vraagKleur(x0, y1)) n++;
        if (vraagKleur(x, y1)) n++;
        if (vraagKleur(x1, y1)) n++;
        return n;
    }
}